enhancements/project-request-limit.md

```diff
@@ -0,0 +1,293 @@
+ pushed to OperatorHub independently of OpenShift release cycle.
+3. How do we describe version skew limitations to OLM so our operator gets uninstalled *before* an illegal downgrade or
+ upgrade? This is a concrete case of the API we want to use isn't available before 1.16 and after 1.18, the previous API could be gone.
+
+## Proposal
+1. Create a `Validating` admission webhook server that provides `requestlimit.project.openshift.io/ProjectRequestLimit`.
+2. The admission webhook will enable `localhost` serving and `kube-apiserver` will talk to it using a `localhost` URL on the host network.
+2. Create an operator that manages all lifecycle aspects of the admission webhook via a custom resource.
+3. Productize the operator and ship it as a `RedHat operator` via `OperatorHub`. 
+4. Provide official documentation on how to interact with the `ProjectRequestLimit` operator.
+
+### User Stories [optional]
+
+*Story 1*: As an engineer I want to port `ProjectRequestLimit` plugin from 3.11 into a `Validating` admission webhook. 
+
+Notes:
+* Use  [generic-admission-server](https://github.com/openshift/generic-admission-server) to wire the validating admission webhook server.
+* The source code is here: https://github.com/openshift/origin/blob/release-3.11/pkg/project/apiserver/admission/requestlimit/admission.go 
+
+*Story 2*: As an engineer I want to write an operator that manages the `ProjectRequestLimit` admission webhook server.
+
+*Story 3*: As an engineer I want to ship `ProjectRequestLimit` as a RedHat operator via OperatorHub.
+
+Notes:
+* The operator is OLM enabled.
+* For productization, if we go through ART, integration work should target 4.5.
+* The operator should support disconnected install.
+  
+
+### Implementation Details/Notes/Constraints [optional]
+We *must* be able to re-bootstrap the cluster. This means that a cluster with this admission plugin created must be able
+to be completely shut down and subsequently restarted.
+
+#### The Operand
+The `Validating` admission webhook will implement the following API.
+```
+Group: requestlimit.project.openshift.io
+Version: v1
+Resource: requests
+``` 
+
+The webhook will not be exposed via any `Service`, on the other hand, `kube-apiserver` will `POST` 
+`AdmissionReview` requests(s) to it directly via the following URL.
+```
+https://{server}/apis/requestlimit.project.openshift.io/v1/requests
+```
+
+The webhook will implement `localhost` serving, as follows:
+* The webhook server should be hosted on the same node as the `kube-apiserver`.
+* The webhook server should be reachable to the `kube-apiserver` over `localhost` of the `hostnetwork`.
+* Communication between `kube-apiserver` and the webhook server must be secure using `x509`. 
+* The webhook server should only allow connection over `localhost`, thus it would disallow connection(s) originating 
+  from outside the host.
```

Comment:
I miss why we want to communicate via localhost.

Reply:
localhost serving is no longer applicable, the webhook will be fronted by a `Service`